Spot check bug throws internal_error when record number is multiple of spot check number

**Describe the bug**
internal_error thrown from Validate with not much information provided:

```
PDS4 Bundle Level Validation Results

  FAIL: file:/mnt/pdsdata/nsyt/delivery/latest-release/insight_cameras/data/sol/0012/mipl/rdr/mesh/D032L0012_597604799CPG_F0101I0060M1.obj.xml
      ERROR  [error.validation.internal_error]   Uncaught exception while validating: null

  FAIL: file:/mnt/pdsdata/nsyt/delivery/latest-release/insight_cameras/data/sol/0012/mipl/rdr/mesh/D033L0012_597605011CPG_F0101I0060M1.obj.xml
      ERROR  [error.validation.internal_error]   Uncaught exception while validating: null

  FAIL: file:/mnt/pdsdata/nsyt/delivery/latest-release/insight_cameras/data/sol/0250/mipl/rdr/mesh/D001R0250_618729198CWG_F0101_0028M6.obj.xml
      ERROR  [error.validation.internal_error]   Uncaught exception while validating: null
```

After pulling the tagged release and testing, it looks like the error raised is actually in PDS4-JParser:
```
Caused by: java.lang.IllegalArgumentException: The index is out of range 1 - 119000
	at gov.nasa.pds.objectAccess.TableReader.getRecord(TableReader.java:182)
	at gov.nasa.pds.tools.validate.content.table.RawTableReader.getRecord(RawTableReader.java:177)
	at gov.nasa.pds.tools.validate.rule.pds4.TableDataContentValidationRule.validateTableDataContents(TableDataContentValidationRule.java:332)
	... 60 more
```

--- a/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableDataContentValidationRule.java
+++ b/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableDataContentValidationRule.java
@@ -371,24 +371,28 @@ public class TableDataContentValidationRule extends AbstractValidationRule {
                     tableIndex,
                     reader.getCurrentRow());                
               }
+              
+              if (spotCheckData > 1) {
+                  // If spot checking is turned on, we want to skip to
+                  // the record just before the one we want to read
+                  try {
+                    record = reader.getRecord(reader.getCurrentRow() + (spotCheckData - 1));
+                  } catch (IllegalArgumentException iae) {
+                    line = null;
+                    continue;
+                  } catch (IOException io) {
+                    throw new IOException("Error occurred "
+                        + "while reading table '" + tableIndex + "', "
+                        + "record '"
+                        + (reader.getCurrentRow() + spotCheckData) + "'");
+                  }
+              }
+              
+              // WARNING: All checks must happen before this in case we
+              // reach end of the table
               if (table instanceof TableDelimited && 
                   definedNumRecords == reader.getCurrentRow()) {
                 break;
-              }
-              if (spotCheckData > 1) {
-                // If spot checking is turned on, we want to skip to
-                // the record just before the one we want to read
-                try {
-                  record = reader.getRecord(reader.getCurrentRow() + (spotCheckData - 1));
-                } catch (IllegalArgumentException iae) {
-                  line = null;
-                  continue;
-                } catch (IOException io) {
-                  throw new IOException("Error occurred "
-                      + "while reading table '" + tableIndex + "', "
-                      + "record '"
-                      + (reader.getCurrentRow() + spotCheckData) + "'");
-                }
               }
               line = reader.readNextLine();            
             }

